File: Banco.java
import java.awt.HeadlessException;
import java.sql.*;
import javax.swing.JOptionPane;

public class Banco {
    public static void main(String[] args) {
        String url = "jdbc:sqlite:menu_banco.db";

        try {
            // Criar tabela inicial
            try (Connection conexao = DriverManager.getConnection(url)) {
                // Criar tabela inicial
                String sqlTabela = "CREATE TABLE IF NOT EXISTS usuarios (" +
                        "id INTEGER PRIMARY KEY AUTOINCREMENT," +
                        "nome TEXT NOT NULL, telefone TEXT NOT NULL, data TEXT NOT NULL);";
                conexao.createStatement().execute(sqlTabela);
                
                boolean continuar = true;
                
                while (continuar) {
                    // Menu de Opções
                    String menu = """
                                                  === SISTEMA DE CADASTRO ===
                                                  1. Cadastrar Usu\u00e1rio
                                                  2. Listar Usu\u00e1rios
                                                  3. Excluir Usu\u00e1rio (por ID)
                                                  4. Sair
                                                                       Escolha uma op\u00e7\u00e3o:""";
                    
                    String opcao = JOptionPane.showInputDialog(null, menu);
                    
                    if (opcao == null || opcao.equals("4")) {
                        break;
                    }
                    
                    switch (opcao) {
                        case "1" -> // CADASTRAR
                            cadastrarUsuario(conexao);
                        case "2" -> // LISTAR
                            listarUsuarios(conexao);
                        case "3" -> // EXCLUIR
                            excluirUsuario(conexao);
                        default -> JOptionPane.showMessageDialog(null, "Opção inválida!");
                    }
                }
            }

        } catch (HeadlessException | SQLException e) {
            JOptionPane.showMessageDialog(null, "Erro: " + e.getMessage());
        }
    }

    // --- MÉTODOS AUXILIARES ---

    private static void cadastrarUsuario(Connection conexao) throws SQLException {
        String nome = JOptionPane.showInputDialog("Nome:");
        String tel = JOptionPane.showInputDialog("Telefone:");
        String data = JOptionPane.showInputDialog("Data (DD/MM/AAAA):");

        if (nome != null && tel != null && data != null) {
            String sql = "INSERT INTO usuarios (nome, telefone, data) VALUES (?, ?, ?)";
            PreparedStatement pstmt = conexao.prepareStatement(sql);
            pstmt.setString(1, nome);
            pstmt.setString(2, tel);
            pstmt.setString(3, data);
            pstmt.executeUpdate();
            JOptionPane.showMessageDialog(null, "Usuário cadastrado!");
        }
    }

    private static void listarUsuarios(Connection conexao) throws SQLException {
        ResultSet rs = conexao.createStatement().executeQuery("SELECT * FROM usuarios");
        StringBuilder lista = new StringBuilder("=== HISTÓRICO ===\n");
        while (rs.next()) {
            lista.append(rs.getInt("id")).append(" - ")
                 .append(rs.getString("nome")).append(" | ")
                 .append(rs.getString("telefone")).append("\n");
        }
        JOptionPane.showMessageDialog(null, lista.length() > 20 ? lista : "Nenhum registro encontrado.");
    }

    private static void excluirUsuario(Connection conexao) throws SQLException {
        String idStr = JOptionPane.showInputDialog("Digite o ID do usuário que deseja excluir:");
        if (idStr != null) {
            String sql = "DELETE FROM usuarios WHERE id = ?";
            PreparedStatement pstmt = conexao.prepareStatement(sql);
            pstmt.setInt(1, Integer.parseInt(idStr));
            int rows = pstmt.executeUpdate();
            if (rows > 0) {
                JOptionPane.showMessageDialog(null, "Usuário removido com sucesso!");
            } else {
                JOptionPane.showMessageDialog(null, "ID não encontrado.");
            }
        }
    }
}

File: banco.py
import json
import os
from datetime import datetime

class Usuario:
    def __init__(self, nome, senha):
        self.nome = nome
        self.senha = senha
        self.data_criacao = datetime.now().strftime("%d/%m/%Y %H:%M:%S")
    
    def to_dict(self):
        return {
            "nome": self.nome,
            "senha": self.senha,
            "data_criacao": self.data_criacao
        }

class BancoDados:
    def __init__(self, arquivo="usuarios.json"):
        self.arquivo = arquivo
        self.usuarios = self._carregar()
    
    def _carregar(self):
        if os.path.exists(self.arquivo):
            with open(self.arquivo, "r", encoding="utf-8") as f:
                return json.load(f)
        return {}
    
    def _salvar(self):
        with open(self.arquivo, "w", encoding="utf-8") as f:
            json.dump(self.usuarios, f, ensure_ascii=False, indent=2)
    
    def cadastrar(self, nome, senha):
        if nome in self.usuarios:
            return False, "Usuário já existe"
        
        usuario = Usuario(nome, senha)
        self.usuarios[nome] = usuario.to_dict()
        self._salvar()
        return True, "Usuário cadastrado com sucesso"
    
    def excluir(self, nome):
        if nome not in self.usuarios:
            return False, "Usuário não encontrado"
        
        del self.usuarios[nome]
        self._salvar()
        return True, "Usuário excluído com sucesso"
    
    def listar(self):
        return self.usuarios

class Painel:
    def __init__(self):
        self.db = BancoDados()
    
    def exibir_menu(self):
        print("\n" + "="*40)
        print("PAINEL DE CONTROLE - BANCO DE DADOS")
        print("="*40)
        print("1. Cadastrar usuário")
        print("2. Excluir usuário")
        print("3. Listar usuários")
        print("4. Sair")
        print("="*40)
    
    def executar(self):
        while True:
            self.exibir_menu()
            opcao = input("Escolha uma opção: ").strip()
            
            if opcao == "1":
                self._cadastrar()
            elif opcao == "2":
                self._excluir()
            elif opcao == "3":
                self._listar()
            elif opcao == "4":
                print("Saindo...")
                break
            else:
                print("Opção inválida!")
    
    def _cadastrar(self):
        nome = input("Nome do usuário: ").strip()
        if not nome:
            print("Nome não pode ser vazio!")
            return
        
        senha = input("Senha: ").strip()
        if not senha:
            print("Senha não pode ser vazia!")
            return
        
        sucesso, mensagem = self.db.cadastrar(nome, senha)
        print(mensagem)
    
    def _excluir(self):
        nome = input("Nome do usuário a excluir: ").strip()
        sucesso, mensagem = self.db.excluir(nome)
        print(mensagem)
    
    def _listar(self):
        usuarios = self.db.listar()
        if not usuarios:
            print("Nenhum usuário cadastrado")
            return
        
        print("\n--- USUÁRIOS CADASTRADOS ---")
        for nome, dados in usuarios.items():
            print(f"Nome: {nome} | Criado em: {dados['data_criacao']}")

if __name__ == "__main__":
    painel = Painel()
    painel.executar()